ginkgo.Jenkinsfile

```diff
@@ -52,68 +51,185 @@ pipeline {
-                            sh 'cd ${TESTDIR}; K8S_VERSION=1.14 ginkgo --focus=" K8s*" -v --failFast=${FAILFAST} -- -cilium.provision=false'
-                        },
-                        failFast: "${FAILFAST}".toBoolean()
-                    )
+            failFast true
+            parallel {
+                stage('BDD-Test-PR-runtime') {
+                    environment {
+                        GOPATH="${WORKSPACE}/runtime-gopath"
+                        TESTDIR="${GOPATH}/${PROJ_PATH}/test"
+                    }
+                    steps {
+                        sh 'cd ${TESTDIR}; ginkgo --focus=" Runtime*" -v --failFast=${FAILFAST} -- -cilium.provision=false'
+                    }
+                    post {
+                        always {
+                            sh 'cd ${TESTDIR}; ./post_build_agent.sh || true'
+                            sh 'cd ${TESTDIR}; ./archive_test_results.sh || true'
+                            sh 'cd ${TESTDIR}/..; mv *.zip ${WORKSPACE} || true'
+                            sh 'cd ${TESTDIR}; mv *.xml ${WORKSPACE}/${PROJ_PATH}/test || true'
+                            sh 'cd ${TESTDIR}; vagrant destroy -f || true'
+                        }
+                        failure {
+                            script {
+                                if  (!currentBuild.displayName.contains('fail')) {
+                                    currentBuild.displayName = 'Runtime tests fail\n' + currentBuild.displayName
+                                }
+                            }
+                        }
+                    }
+                }
+                stage('BDD-Test-PR-K8s-1.10-net-next') {
+                    environment {
+                        GOPATH="${WORKSPACE}/k8s-lower-gopath"
+                        TESTDIR="${GOPATH}/${PROJ_PATH}/test"
+                        NETNEXT="true"
+                    }
+                    steps {
+                        sh 'cd ${TESTDIR}; K8S_VERSION=1.10 ginkgo --focus=" K8s*" -v --failFast=${FAILFAST} -- -cilium.provision=false'
+                    }
+                    post {
+                        always {
+                            sh 'cd ${TESTDIR}; ./post_build_agent.sh || true'
+                            sh 'cd ${TESTDIR}; ./archive_test_results.sh || true'
+                            sh 'cd ${TESTDIR}/..; mv *.zip ${WORKSPACE} || true'
+                            sh 'cd ${TESTDIR}; mv *.xml ${WORKSPACE}/${PROJ_PATH}/test || true'
+                            sh 'cd ${TESTDIR}; K8S_VERSION=1.10 vagrant destroy -f || true'
+                        }
+                        failure {
+                            script {
+                                if  (!currentBuild.displayName.contains('fail')) {
+                                    currentBuild.displayName = 'K8s 1.10-net-next fail\n' + currentBuild.displayName
```

Comment:
What will happen when multiple stages fail that are running in parallel in terms of setting the displayName?

Reply:
From @nebril's comment on setting the displayName, this will probably race and the best case is that we see multiple failure strings, or just last-one-wins.

@nebril I guess if we see this racing we can do this step in the post->failure of the next level up. What I don't know is if we have a way to communicate information in a non-racey way. The best I can think of is making a file but I don't know if we can check those in these steps.

Reply:
@ianvernon usually one build fails and because we are failing fast, other builds are getting an interrupt signal which will cause them to fail or abort - but we don't care about these. That's why we are checking whether the word "fail" is in the name already - after first failure, all other ones are treated as if they occurred because of failfast mechanism.